app/actors/hyrax/actors/asset_actor.rb

```diff
@@ -117,7 +117,9 @@ def annotation_attributes
         end
 
         def find_or_create_admin_data(env)
-          return env.curation_concern.admin_data if env.curation_concern.admin_data.present?
+          # Don't use present? here. It is broken for some ActiveRecord objects. Maybe due to overriding getter
+          # See: https://stackoverflow.com/questions/35410668/what-determines-the-return-value-of-present
+          return env.curation_concern.admin_data if env.curation_concern.admin_data.is_a?(AdminData)
```

Comment:
@bkiahstroud was `present?` returning false positive or false negative here? And do you know what the value of `env.curation_concern.admin_data` was in that circumstance?

I ask because from a business logic perspective, `env.curation_concern.admin_data` should never be anything other than `AdminData` instance or `nil`, so if it's winding up in a 3rd state, that might be an error state we need to look into preventing.

Reply:
It was returning a false negative. `env.curation_concern.admin_data` was returning an instance of `AdminData`. 

```ruby
env.curation_concern.admin_data
=> #<AdminData id: ...
env.curation_concern.admin_data.present?
=> false
```